apps/rush-lib/src/cli/RushPnpmCommandLine.ts

```diff
@@ -0,0 +1,259 @@
+    [/* synonym */ 'i', CommandKind.Blocked],
+    ['install-test', CommandKind.Blocked],
+    [/* synonym */ 'it', CommandKind.Blocked],
+    ['link', CommandKind.ShowWarning],
+    [/* synonym */ 'ln', CommandKind.ShowWarning],
+    ['list', CommandKind.KnownSafe],
+    [/* synonym */ 'ls', CommandKind.KnownSafe],
+    ['outdated', CommandKind.KnownSafe],
+    ['pack', CommandKind.KnownSafe],
+    ['prune', CommandKind.KnownSafe],
+    ['publish', CommandKind.KnownSafe],
+    ['rebuild', CommandKind.KnownSafe],
+    [/* synonym */ 'rb', CommandKind.KnownSafe],
+    ['remove', CommandKind.ShowWarning],
+    [/* synonym */ 'rm', CommandKind.ShowWarning],
+    ['root', CommandKind.KnownSafe],
+    ['run', CommandKind.KnownSafe],
+    ['start', CommandKind.KnownSafe],
+    ['store', CommandKind.KnownSafe],
+    ['test', CommandKind.KnownSafe],
+    [/* synonym */ 't', CommandKind.KnownSafe],
+    ['unlink', CommandKind.ShowWarning],
+    ['update', CommandKind.ShowWarning],
+    [/* synonym */ 'up', CommandKind.ShowWarning]
+  ]);
+
+  public static launch(launcherVersion: string, options: ILaunchRushPnpmInternalOptions): void {
+    // Node.js can sometimes accidentally terminate with a zero exit code  (e.g. for an uncaught
+    // promise exception), so we start with the assumption that the exit code is 1
+    // and set it to 0 only on success.
+    process.exitCode = 1;
+
+    try {
+      // Are we in a Rush repo?
+      let rushConfiguration: RushConfiguration | undefined = undefined;
+      if (RushConfiguration.tryFindRushJsonLocation()) {
+        rushConfiguration = RushConfiguration.loadFromDefaultLocation({ showVerbose: true });
+      }
+
+      NodeJsCompatibility.warnAboutCompatibilityIssues({
+        isRushLib: true,
+        alreadyReportedNodeTooNewError: !!options.alreadyReportedNodeTooNewError,
+        rushConfiguration
+      });
+
+      if (!rushConfiguration) {
+        throw new Error(
+          'The "rush-pnpm" command must be executed in a folder that is under a Rush workspace folder'
+        );
+      }
+
+      if (rushConfiguration.packageManager !== 'pnpm') {
+        throw new Error(
+          'The "rush-pnpm" command requires your rush.json to be configured to use the PNPM package manager'
+        );
+      }
+
+      if (!rushConfiguration.pnpmOptions.useWorkspaces) {
+        throw new Error(
+          'The "rush-pnpm" command requires the "useWorkspaces" setting to be enabled in rush.json'
+        );
+      }
+
+      const workspaceFolder: string = rushConfiguration.commonTempFolder;
+      const workspaceFilePath: string = path.join(workspaceFolder, 'pnpm-workspace.yaml');
+
+      if (!FileSystem.exists(workspaceFilePath)) {
+        console.error(colors.red('Error: The PNPM workspace file has not been generated:'));
+        console.error(`  ${colors.red(workspaceFilePath)}\n`);
+        console.error(colors.cyan(`Do you need to run "rush install" or "rush update"?`));
```

Comment:
Should we consider running `rush install` on the dev's behalf? I had thought about doing a half-install where we would do all our setup/configuration but don't actually install, since that would be all that's needed in many cases... but that seemed like a bad plan (allowing devs to get into a semi-installed state)

Reply:
Possibly. But we would need to implement logic for tracking whether `rush install` needs to be run or not. Which is related to another feature request, that `rush build` should fail if Rush detects that `rush install` was not run.

If we implement that, it should be in a separate PR.